src/basic/cgroup-util.c

```diff
@@ -471,6 +471,64 @@ int cg_kill_recursive(
         return ret;
 }
 
+int cg_kill_all(void) {
+        _cleanup_closedir_ DIR *d = NULL;
+        char *cg_root;
+        int r;
+
+        if (cg_all_unified() <= 0)
+                return -EOPNOTSUPP;
+
+        r = cg_get_root_path(&cg_root);
+        if (r < 0)
+                return r;
+
+        r = cg_enumerate_subgroups(SYSTEMD_CGROUP_CONTROLLER, cg_root, &d);
+        if (r < 0)
+                return r;
+
+        for (;;) {
+                _cleanup_free_ char *node = NULL, *path = NULL;
+
+                r = cg_read_subgroup(d, &node);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
```

Comment:
Not a matter, but
```c
if (r <= 0)
        return r;
```
is enough, then you can drop `return 0;` at the end of the function.

Reply:
it looks cleaner to break out to me

Reply:
No problem. Then, feel free to ignore.